pandas/src/datetime/np_datetime_strings.c

```diff
@@ -460,7 +460,7 @@ parse_iso_8601_datetime(char *str, int len,
         }
 
         /* Check the casting rule */
-        if (unit != -1 && !can_cast_datetime64_units(bestunit, unit,
```

Comment:
I guess checking `-1` or not is for https://github.com/pydata/pandas/blob/820e1105a44f6f9347a02ebed0a85b4b71d70363/pandas/src/datetime/np_datetime_strings.c#L316 .
Only this line assigns `-1` for `PANDAS_DATETIMEUNIT ` a type variable.